src/Tests/Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests/ParameterCapturing/MethodSignatureTests.cs

```diff
@@ -0,0 +1,280 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using SampleMethods;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests.ParameterCapturing
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    public class MethodSignatureTests
+    {
+        [Theory]
+        [MemberData(nameof(GetMethodSignatureTestData))]
+        public void MethodSignature(
```

Comment:
IMO the `[MemberData(nameof(GetMethodSignatureTestData))]` is harder to identify the cases being tested, even looking at the method it takes some mental overhead to figure out what each data set is trying to test.

Let's refactor this method into one called `TestCore` that does the AAA, and each set of data into a `[Fact]` test method with a descriptive name of what is being tested